lib/kdl/include/kdl/result.h

```diff
@@ -77,8 +90,8 @@ namespace kdl {
         }
         
         /**
-         * Applies the given visitor to the given result result and returns the result returned by the visitor.
-         * The value or error contained in the given result result is passed to the visitor by const lvalue reference.
+         * Applies the given visitor to the given result and returns the result returned by the visitor.
+         * The value or error contained in the given result is passed to the visitor by const lvalue reference.
```

Comment:
I think this doc needs some improvement

The key bits of info that are important here:
- visitor must be able to handle all cases of the result (success and error cases)
- the visitor must return the same type for all possible cases (they give a more technical wording https://en.cppreference.com/w/cpp/utility/variant/visit but that seems good enough)
- the visit_result function returns the return value of the visitor as-is (in constrast to map_result)

So I would word this as something like:

Applies a visitor which can accept the success type and all error types of this result. Returns the return value of the visitor.
The visitor must return the same type for all possible cases.

Reply:
I'll make these changes in a later PR: https://github.com/kduske/TrenchBroom/issues/2983#issuecomment-658147940